storagepool/fileBasedHealthCheck.go

```diff
@@ -0,0 +1,38 @@
+package storagepool
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type fileBasedHealthCheck struct {
+	healthCheckBaseDir string
+}
+
+func (fc *fileBasedHealthCheck) populateHostMap() (map[string]string, error) {
+	activeHosts := map[string]string{}
+	f, err := os.Open(fc.healthCheckBaseDir)
+	if err != nil {
+		return nil, err
+	}
+	fi, err := f.Readdir(-1)
```

Comment:
It does the same thing that im doing -    99	// a list of sorted directory entries.
```
   100	func ReadDir(dirname string) ([]os.FileInfo, error) {
   101		f, err := os.Open(dirname)
   102		if err != nil {
   103			return nil, err
   104		}
   105		list, err := f.Readdir(-1)
   106		f.Close()
   107		if err != nil {
   108			return nil, err
   109		}
   110		sort.Sort(byName(list))
   111		return list, nil
   112	}
   113	
```

I'll just add f.Close()?